doc/howtos/bluetoothspeaker.rst

```diff
@@ -0,0 +1,141 @@
+Bluetooth Speaker
+#################
+
+How to setup your intel-iot-refkit image to function as a bluetooth speaker
+===========================================================================
+
+For configuring the bluetooth speaker setup you should use the
+"gateway" image and prepare couple of configuration files. This
+configuration has been tested to work with Intel Joule development
+board which has bluetooth radio chip integrated. The following later
+detailed 5 steps should do it:
+
+1. Start ``PulseAudio`` in the boot
+2. Add ``D-Bus`` configuration so that ``PulseAudio`` can communicate with ``BlueZ``
+3. Configure ``BlueZ`` to enable A2DP sink
+4. Unblock bluetooth at boot with ``ConnMan``
+5. Pair and connect your source device
+
+Start ``PulseAudio`` in the boot
+============================
+
+The way to start ``PulseAudio`` in the boot is to use systemd service
+file. In this example service file we start ``PulseAudio`` in system
+mode and don't let the daemon exit when in idle (which is normal
+behavior). ``PulseAudio`` will just go to "sleep"::
+
+  pulseaudio
+  [Unit]
+  Description=Pulse Audio
+
+  [Service]
+  Type=simple
+  ExecStart=/usr/bin/pulseaudio --system --disallow-exit --disable-shm --exit-idle-time=-1
+
+  [Install]
+  WantedBy=multi-user.target
+
+Copy the ``pulseaudio.service`` file to::
+
+  /etc/systemd/system/
+
+and enable it by::
+
+  systemctl enable pulseaudio.service
+
+For this particular use case we are starting ``PulseAudio`` in system
+mode and including bluetooth modules to the system.pa file. Remember
+that starting ``PulseAudio`` in system mode is not the recommended way
+and you should be aware of the possible consequences.
+
+system.pa can be found from::
+
+  /etc/pulse/
+
+Add the following lines to the system.pa to enable bluetooth::
+
+  .ifexists module-bluetooth-policy.so
+  load-module module-bluetooth-policy
+  .endif
+
+  .ifexists module-bluetooth-discover.so
+  load-module module-bluetooth-discover
+  .endif
+
+Add ``D-Bus`` configuration so that ``PulseAudio`` can communicate with ``BlueZ``
+=============================================================================
+
+Copy ``D-Bus`` configuration file (name it for example as
+pulseaudio-bluetooth.conf) to::
+
+  /etc/dbus-1/system.d/
+
+ Contents of the file should be the following::
+
+   <busconfig>
+     <policy user="pulse">
+       <allow send_destination="org.bluez"/>
+     </policy>
+   </busconfig>
+
+Configure ``BlueZ`` to enable A2DP sink
+=======================================
+
+``BlueZ`` needs configuration to enable A2DP sink feature.
+This can be done by copying file audio.conf to::
+
+  /etc/bluetooth/
+
+With following contents::
+
+  [General]
+  Enable=Source
+
+Unblock bluetooth radio at boot with ``ConnMan``
+================================================
+
+By default the bluetooth radio is soft blocked. To enable it in the
```

Comment:
there are concepts "hard" and "soft" blocked (actually don't know who defines these, rfkill etc.). Hard blocked means powered off, soft blocked means the radio is powered but its not "online" so you can't connect to it. As a safety feature in linux distros, radios are usually at least soft blocked when booting the computer. In bluetooth speaker however it doesn't make too much sense.